public/app/features/alerting/unified/utils/rule-form.ts

```diff
@@ -431,3 +433,23 @@ export function getIntervals(range: TimeRange, lowLimit?: string, resolution?: n
 
   return rangeUtil.calculateInterval(range, resolution, lowLimit);
 }
+
+export function fixBothInstantAndRangeQuery(query: AlertQuery) {
+  const model = query.model;
+
+  if (!isPromQuery(model)) {
+    return query;
+  }
+
+  // I'm very sorry but I don't know how to start refactoring this to use Generics
+  const isBothInstantAndRange = model.instant && model.range;
+  if (isBothInstantAndRange) {
+    return { ...query, model: { ...model, range: true, instant: false } };
+  }
+
+  return query;
+}
+
+function isPromQuery(model: AlertDataQuery): model is PromQuery {
+  return 'expr' in model && 'instant' in model && 'range' in model;
```

Comment:
Can we be sure this properties are exclusive to Prometheus query?

Reply:
I checked the other datasources and so far it appears to exist for prometheus and loki query – though only Prometheus supports "both". This doesn't affect the code though as this type guard is only really necessary to make sure `range` and `instant` are mutually exclusive.